librz/bin/bin.c

```diff
@@ -958,6 +958,56 @@ RZ_API const char *rz_bin_entry_type_string(int etype) {
 	return NULL;
 }
 
+/**
+ * \brief Returns an entry point address to seek to after a binary was loaded.
+ * If no entry point could be determined, it just returns 0, which is also a valid address.
+ *
+ * The entry point is determined in the following order:
+ * 1.  'entry0' flag is not NULL? --> Address of entry0.
+ * 2. Object is NULL?             --> Return 0.
+ * 3. Object provides entries?    --> Return first entry point of object.
+ * 4. Object has sections?        --> Return address of first executable segment.
+ * 5. Otherwise?                  --> Return 0.
+ *
+ * It always returns virtual addresses if one is defined.
+ *
+ * \param core The core to use.
+ * \param obj The object file to get the entry point from.
+ * \param entry0 The flag for the entry0 symbol if any.
+ *
+ * \return The entry point address of the binary.
+ */
+RZ_API ut64 rz_bin_get_entrypoint(RZ_NONNULL RZ_BORROW RzCore *core, RZ_NULLABLE RzBinObject *obj, RZ_NULLABLE RzFlagItem *entry0) {
+	rz_return_val_if_fail(core, 0);
+	if (entry0) {
+		// entry0 symbol present. This is kept as default seek if present.
+		// Simply due to compatibility.
+		return entry0->offset;
+	} else if (!obj) {
+		return 0;
+	} else if (obj->entries && rz_pvector_len(obj->entries) > 0) {
+		// The binary loader specified entry points. Use the first one.
+		const RzBinAddr *entry = rz_pvector_at(obj->entries, 0);
+		ut64 addr = entry->vaddr ? entry->vaddr : entry->paddr;
+		return addr;
+	}
+	const RzPVector *sections = rz_bin_object_get_sections_all(obj);
+	if (!sections) {
+		return 0;
+	}
+	// The binary loader did not specify entry points.
+	// Fall back to the first executable section.
+	void **iter;
+	rz_pvector_foreach (sections, iter) {
+		RzBinSection *s = *iter;
+		if (s->perm & RZ_PERM_X) {
+			ut64 addr = s->vaddr ? s->vaddr : s->paddr;
+			return addr;
+		}
+	}
+	return 0;
+}
```

Comment:
if is `rz_bin_` then it cannot have `RzCore`.

Reply:
i feel like this should be `RZ_API bool rz_bin_object_get_first_entrypoint(RZ_NONNULL RzBinObject *obj) {`